### DISCUSSION TITLE
Is OpenFF 2.0 compatible with AMBER ff19sb? and if yes how to combine the protein and ligand prmtop file?

### DISCUSSION THREAD

**Discussion by aniketsh** [openforcefield/discussions#8] [Q&A] [answered]
Is OpenFF 2.0 compatible with AMBER ff19sb? and if yes how to combine the protein and ligand prmtop file?

**Comment by j-wags** [comment, marked as answer]
Hi @aniketsh,

> Is OpenFF 2.0 compatible with AMBER ff19sb? 

I believe that they are compatible, since both `ff19sb` and `openff-2.0.0` were fit using similar methods to a mix of QM and physical property data, and both force fields use compatible functional forms.

> if yes how to combine the protein and ligand prmtop file?

This is the hard part - My understanding is that we generally recommend use of ff14sb or earlier, since ff19sb introduces "CMAP" terms which aren't consistently supported in other force field definitions/simulation engines. I think that OpenMM may be able to handle them, but don't have first-hand experience. So `sander`/`pmemd` may be the best simulation engines of choice to ensure that the terms are interpreted correctly.

Generally, there are two options for making systems that use a mix of multiple force fields:
1) Use `openmmforcefields` to assign parameters to everything, then export the resulting OpenMM `System` to a ParmEd `Structure` and convert that to your format of choice
2) Assemble the different components using the tool that's best suited for the job, and then combine them using ParmEd. 

Since [`openmmforcefields` doesn't seem to support ff19sb yet](https://github.com/openmm/openmmforcefields/issues/75), I think that approach 2 is the only option right now. It would look something like [this example](https://github.com/openforcefield/openff-toolkit/blob/36d286e7af2628cb56d5945d74ecd0673e6d84f7/examples/using_smirnoff_with_amber_protein_forcefield/BRD4_inhibitor_benchmark.ipynb)

* parameterize the protein using tleap, writing the result  to `protein.prmtop` and `protein.inpcrd` files
* parameterize the ligand using OpenFF, getting an OpenMM `System` object as a result (you can either do the solvent and ions using tleap or the OpenFF toolkit, though note that openff-2.0.0 is trained to be compatible with TIP3P water. The toolkit will apply TIP3P parameters for water as part of the `openff-2.0.0` force field, but if you have tleap handle the water be sure to specify TIP3P)
* turn the OpenMM `System` into a ParmEd `Structure`
* loading `protein.prmtop` and `protein.inpcrd` into another ParmEd `Structure`
* Use ParmEd's `+` operator to make the complex: `complex_structure = protein_structure + ligand_structure`

I'd note, however, that you should look **very closely** at the results of this conversion. All of the CMAP-handling functionality in OpenMM and ParmEd seems to be just a few months old, so there may be conversion/interpretation bugs that people haven't found yet!

**Comment by aniketsh** [comment]
Thanks Jeff. I use parmed usually to combine the system and it had worked before. With CMAP though it works in only one direction. 

Parmed version (3.4.3) 

import parmed as pmd

ptn=pmd.amber.AmberParm(prm_name='host.prmtop',rst7_name='host.crd')
lig=pmd.amber.AmberParm(prm_name='MOL.prmtop',rst7_name='MOL.inpcrd')

system=ptn+lig # This works 
system=lig+ptn # However this doesn't 

system.save('sys.prmtop',overwrite=True)
system.save('sys.crd',overwrite=True)

**Comment by mattwthompson** [comment]
system=ptn+lig # This works
    system=lig+ptn # However this doesn't

This looks like a ParmEd bug of some sort, could you open an issue there?
